ipa-core/src/protocol/ipa_prf/boolean_ops/share_conversion_aby.rs

```diff
@@ -126,84 +132,103 @@ where
+
+    for (record_id, x) in record_id_range.clone().zip(input_shares) {
+        // generate sh_r = (0, 0, sh_r) and sh_s = (sh_s, 0, 0)
+        // the two highest bits are set to 0 to allow carries for two additions
+        let (sh_r, sh_s) = gen_sh_r_and_sh_s::<_, BITS, NC>(ctx_gen, RecordId::from(record_id));
+
+        // addition r+s might cause carry,
+        // this is no problem since we have set bit 254 of sh_r and sh_s to 0
+        let sh_rs = {
+            let (mut rs_with_higherorderbits, _) =
+                integer_add::<_, TwoHundredFiftySixBitOpStep, NC>(
+                    ctx_masks.clone(),
+                    RecordId::from(record_id),
+                    &sh_r,
+                    &sh_s,
+                )
+                .await?;
+
+            // PRSS/Multiply masks added random highest order bit,
+            // remove them to not cause overflow in second addition (which is mod 256):
+            rs_with_higherorderbits[BITS - 1] = AdditiveShare::<Boolean, NC>::ZERO;
+
+            // return rs
+            rs_with_higherorderbits
+        };
+
+        // addition x+rs, where rs=r+s might cause carry
+        // this is not a problem since bit 255 of rs is set to 0
+        let (sh_y, _) = integer_add::<_, TwoHundredFiftySixBitOpStep, NC>(
+            ctx_mask_x.clone(),
+            RecordId::from(record_id),
+            &sh_rs,
+            &x,
         )
         .await?;
-
-        // PRSS/Multiply masks added random highest order bit,
-        // remove them to not cause overflow in second addition (which is mod 256):
-        rs_with_higherorderbits[BITS - 1] = AdditiveShare::<Boolean, NC>::ZERO;
-
-        // return rs
-        rs_with_higherorderbits
-    };
-
-    // addition x+rs, where rs=r+s might cause carry
-    // this is not a problem since bit 255 of rs is set to 0
-    let (sh_y, _) = integer_add::<_, TwoHundredFiftySixBitOpStep, NC>(
-        m_ctx.narrow(&Step::IntegerAddMaskToX),
-        record_id,
-        &sh_rs,
-        &x,
-    )
-    .await?;
+        vec_sh_y.push(sh_y);
+        vec_sh_s.push(sh_s);
+        vec_sh_r.push(sh_r);
+    }
 
     // validate before reveal
-    validator.validate().await?;
-
-    // this leaks information, but with negligible probability
-    let mut y = (m_ctx.role() != Role::H3).then(|| Vec::with_capacity(NC));
-    for i in 0..BITS {
-        let y_bit = partial_reveal(
-            m_ctx.narrow(&Step::RevealY(i)),
-            record_id,
-            Role::H3,
-            sh_y.get(i).unwrap(),
-        )
-        .await?;
-        match (&mut y, y_bit) {
-            (Some(y), Some(y_bit)) => y.push(y_bit),
-            (None, None) => (),
-            _ => unreachable!("inconsistent partial_reveal behavior"),
+    dzkp_validator.validate_chunk(ctx_counter).await?;
+
+    // output vec
+    // NP-vectorized Fp25519 shares
+    let mut results = Vec::<AdditiveShare<Fp25519, NP>>::with_capacity(NC * vec_sh_y.len());
```

Comment:
There was another piece of the suggestion here besides the comment -- the size should be `NC * vec_sh_y.len() / NP`.